File: src/pub_sub/spherical_coord_pub.py
#!/usr/bin/python3

import math
import rospy

from bites_hackathon.msg import spherical_coord

def sph_coord_pub():
    pub = rospy.Publisher('spherical_coord', spherical_coord, queue_size=10)
    rospy.init_node('spherical_coord_pub', anonymous=True)
    rate = rospy.Rate(1) # 10hz
    rad = 1.0
    azi = math.pi/2
    ele = math.pi/2
    for _ in range(10):
        rad = rad + 0.1
        azi = azi + 0.1
        ele = ele + 0.1
        msg = spherical_coord()
        msg.rad = rad
        msg.azi = azi
        msg.ele = ele
        rospy.loginfo(msg)
        pub.publish(msg)
        rate.sleep()

if __name__ == '__main__':
    try:
        sph_coord_pub()
    except rospy.ROSInterruptException:
        pass

File: src/cpp/pub_sub/spherical_coord_pub.cpp

#include "ros/ros.h"
#include "bites_hackathon/spherical_coord.h"
#include <sstream>

int main(int argc, char **argv) 
{
    ros::init(argc, argv, "spherical_coord_pub");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<bites_hackathon::spherical_coord>("spherical_coord", 10);
    ros::Rate rate(1);  // 1 Hz

    double rad = 1.0;
    double azi = M_PI / 2;
    double ele = M_PI / 2;

    for (int i = 0; i < 10; ++i) {
        rad += 0.1;
        azi += 0.1;
        ele += 0.1;

        bites_hackathon::spherical_coord msg;
        msg.rad = rad;
        msg.azi = azi;
        msg.ele = ele;

        ROS_INFO("Publishing spherical coordinate: rad=%.2f, azi=%.2f, ele=%.2f", msg.rad, msg.azi, msg.ele);
        pub.publish(msg);
        rate.sleep();
    }

    return 0;
}